core/src/main/java/org/elasticsearch/index/IndexService.java

```diff
@@ -576,6 +573,18 @@ public synchronized void updateMetaData(final IndexMetaData metadata) {
             if (refreshTask.getInterval().equals(indexSettings.getRefreshInterval()) == false) {
                 rescheduleRefreshTasks();
             }
+            final Translog.Durability durability = indexSettings.getTranslogDurability();
+            if (durability != oldTranslogDurability) {
+                rescheduleFsyncTask(durability);
+            }
+        }
+    }
+
+    private void rescheduleFsyncTask(Translog.Durability durability) {
+        try {
+            IOUtils.closeWhileHandlingException(fsyncTask);
```

Comment:
Hmm can we at least log any exception that came from here instead of suppressing?